src/index.ts

```diff
@@ -2340,15 +2401,15 @@ export default class Decimal {
     }
 
     if (height < 0) {
-      return Decimal.iteratedlog(payload, this, -height);
+      return Decimal.iteratedlog(payload, this, -height, linear);
     }
 
     payload = D(payload);
     const oldheight = height;
     height = Math.trunc(height);
     const fracheight = oldheight - height;
 
-    if (this.gt(Decimal.dZero) && this.lte(1.44466786100976613366)) {
+    if (this.gt(Decimal.dZero) && this.lte(1.44466786100976613366) && (oldheight > 10000 || !linear)) {
```

Comment:
I understand everything else, but what's the spiciness happening around here with the checks to 10000?

Reply:
Basically, if we're using the linear approximation, we want all bases, even those less than e^(1/e), to be accurate. The way you handled bases less than e^(1/e) works fine for higher tetra-exponents, but it was resulting in inaccuracies for lower, non-whole tetra-exponents compared to the linear approximation (it's fine for your analytic tetration). 10000 was the height at which your script stops and does the final approximation, so it's the value I used here for "if the height is greater than this, the approximate way should work". The !linear check ensures that the analytic behavior isn't changed. Basically, the linear approximation can and should just be done the normal way, even on small bases, as long as the height isn't high enough that the values would converge anyway.